File: problems_0-1000/448-FindAllNumbersDisappearedinanArray/python/solution.py
class Solution:
    def findDisappearedNumbers(self, nums: list[int]) -> list[int]:
        set_nums = set(nums)
        list = []
        for i in range(1, len(nums) + 1):
            if i not in set_nums:
                list.append(i)
        return list

if __name__ == "__main__":
    solution = Solution()
    nums = [4,3,2,7,8,2,3,1]
    print(solution.findDisappearedNumbers(nums))

File: problems_0-1000/448-FindAllNumbersDisappearedinanArray/java/Solution.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Solution {
    
    public List<Integer> findDisappearedNumbers(int[] nums) {

        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            set.add(i + 1);
        }
        for (int i = 0; i < nums.length; i++) {
            if (set.contains(nums[i])) {
                set.remove(nums[i]);
            }
        }
        return new ArrayList<>(set);
    } 

    public static void main(String[] args) {
        int[] nums = {4,3,2,7,8,2,3,1};
        Solution solution = new Solution();
        List<Integer> result = solution.findDisappearedNumbers(nums);
        System.out.println(result);

    }
}
